pkg/machine/sockets.go

```diff
@@ -0,0 +1,90 @@
+package machine
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SetSocket creates a new machine file for the socket and assigns it to
+// `socketLoc`
+func SetSocket(socketLoc *VMFile, path string, symlink *string) error {
+	socket, err := NewMachineFile(path, symlink)
+	if err != nil {
+		return err
+	}
+	*socketLoc = *socket
+	return nil
+}
+
+// ReadySocketPath returns the filepath for the ready socket
+func ReadySocketPath(runtimeDir, machineName string) string {
+	return filepath.Join(runtimeDir, fmt.Sprintf("%s_ready.sock", machineName))
+}
+
+// ListenAndWaitOnSocket waits for a new connection to the listener and sends
+// any error back through the channel. ListenAndWaitOnSocket is intended to be
+// used as a goroutine
+func ListenAndWaitOnSocket(errChan chan<- error, connChan chan<- net.Conn, listener net.Listener) {
+	conn, err := listener.Accept()
+	if err != nil {
+		logrus.Error(err)
+		errChan <- err
```

Comment:
Why log the error and return it? If you return it via channel then it should be enough as the caller will likely log or return it so this

Reply:
Done.